e2e/pollers.go

```diff
@@ -73,7 +73,19 @@ func waitUntilPodReady(ctx context.Context, kube *Kubeclient, podName string, t
 			remaining := time.Until(deadline)
 			if currentLogTime.Sub(lastLogTime) > logInterval {
 				// this logs every 5 minutes to reduce spam, iterations of poller are continuning as normal.
-				t.Logf("pod %s status: %s time before timeout: %v", podName, pod.Status.Phase, remaining)
+				t.Logf("-- pod metadata --\n")
```

Comment:
I would just dump the `pod.Status` object

Reply:
imo dumping pod.Status isn't the most readable 

```
Status: {Pending [{PodScheduled False 0001-01-01 00:00:00 +0000 UTC 2024-11-22 14:37:56 -0800 PST Unschedulable 0/8 nodes are available: 2 node(s) didn't match Pod's node affinity/selector, 6 node(s) had untolerated taint {node.cloudprovider.kubernetes.io/uninitialized: true}. preemption: 0/8 nodes are available: 8 Preemption is not helpful for scheduling.}]     []  [] <nil> [] [] BestEffort []  []}
```

instead I've added condition.Reason and condition.Message which was the only other attributes that I think have relevant information
```
    pollers.go:120: -- pod metadata --
    pollers.go:121:    Name: test-pod-abe2e-2024-11-22-tmv6-ubuntu2204gen2containerdairgappedk8000000
                       Namespace: default
                       Node: 
                       Status: Pending
                       Start Time: <nil>
    pollers.go:123:     Reason: Unschedulable
    pollers.go:124:     Message: 0/2 nodes are available: 1 node(s) didn't match Pod's node affinity/selector, 1 node(s) had untolerated taint {node.cloudprovider.kubernetes.io/uninitialized: true}. preemption: 0/2 nodes are available: 2 Preemption is not helpful for scheduling.
```